CRM/Core/DAO/AllCoreTables.php

```diff
@@ -181,12 +181,19 @@ public static function isCoreTable($tableName) {
   /**
    * Get the DAO for a BAO class.
```

Comment:
```suggestion
   * Get the DAO for a BAO class. 
   * (If no parent class is a DAO, returns the ultimate base class.)
```

Reply:
Actually, if no parent class is a DAO then it will return the original `$className` that was passed in.